packages/components/src/palette-edit/index.tsx

```diff
@@ -489,10 +589,16 @@ export default function PaletteEdit( {
 						<ColorPickerPopover
 							isGradient={ isGradient }
 							onClose={ () => setEditingElement( null ) }
-							onChange={ ( newElement ) => {
+							onChange={ (
+								newElement: typeof elements[ number ]
+							) => {
 								debounceOnChange(
+									// @ts-expect-error
 									elements.map(
```

Comment:
`@ts-expect-error` is hacky, but I couldn't find a way around it.

It's for:

<details>
<summary>error message</summary>

```sh
packages/components/src/palette-edit/index.tsx:596:10 - error TS2345: Argument of type '[(Color | Gradient)[]]' is not assignable to parameter of type '[newColors?: Color[] | undefined] | [newGradients?: Gradient[] | undefined]'.
  Type '[(Color | Gradient)[]]' is not assignable to type '[newGradients?: Gradient[] | undefined]'.
    Type '(Color | Gradient)[]' is not assignable to type 'Gradient[]'.
      Type 'Color | Gradient' is not assignable to type 'Gradient'.
        Property 'gradient' is missing in type 'Color' but required in type 'Gradient'.

596          elements.map(
             ~~~~~~~~~~~~~
597           (
    ~~~~~~~~~~~
... 
607           }
    ~~~~~~~~~~~
608          )
    ~~~~~~~~~~

  packages/components/src/palette-edit/types.ts:18:2
    18  gradient: string;
        ~~~~~~~~
    'gradient' is declared here.


Found 1 error.
```

</details>

Reply:
Let's just add some kind of note to signal that it isn't ideal 😄

```tsx
// @ts-expect-error TODO: Don't know how to resolve
```

Reply:
Good idea! Committed verbatim in https://github.com/WordPress/gutenberg/pull/47764/commits/bb3fcdd97db68001578f300a08814bed5cf4a8e2